core/utils/memory_utils.h

```diff
@@ -69,6 +70,37 @@ namespace memoryUtils
     inline constexpr T GB(T bytes) {
       return  bytes / ((T)1024 * (T)1024 * (T)1024);
     }
+
+    /** Returns the current alignment of the ptr.
+     * Essentially just counts the amount of LSB zeros.
+     * Returns 2^31 for ptr == 0. */
+	inline uint32_t alignmentOf(uintptr_t ptr) {
```

Comment:
how about 64b machines?

Reply:
<details>
<summary>Example (click to expand)</summary>

[Generated by this](https://cplayground.com/?p=fox-grasshopper-owl)

```
alignmentOf(1) = 1
alignmentOf(2) = 2
alignmentOf(3) = 1
alignmentOf(4) = 4
alignmentOf(5) = 1
alignmentOf(6) = 2
alignmentOf(7) = 1
alignmentOf(8) = 8
alignmentOf(9) = 1
alignmentOf(10) = 2
alignmentOf(11) = 1
alignmentOf(12) = 4
alignmentOf(13) = 1
alignmentOf(14) = 2
alignmentOf(15) = 1
alignmentOf(16) = 16
alignmentOf(17) = 1
alignmentOf(18) = 2
alignmentOf(19) = 1
alignmentOf(20) = 4
alignmentOf(21) = 1
alignmentOf(22) = 2
alignmentOf(23) = 1
alignmentOf(24) = 8
alignmentOf(25) = 1
alignmentOf(26) = 2
alignmentOf(27) = 1
alignmentOf(28) = 4
alignmentOf(29) = 1
alignmentOf(30) = 2
alignmentOf(31) = 1
alignmentOf(32) = 32
alignmentOf(33) = 1
alignmentOf(34) = 2
alignmentOf(35) = 1
alignmentOf(36) = 4
alignmentOf(37) = 1
alignmentOf(38) = 2
alignmentOf(39) = 1
alignmentOf(40) = 8
alignmentOf(41) = 1
alignmentOf(42) = 2
alignmentOf(43) = 1
alignmentOf(44) = 4
alignmentOf(45) = 1
alignmentOf(46) = 2
alignmentOf(47) = 1
alignmentOf(48) = 16
```

</details>

This function is intended for alignment checking, i.e. something like `assert(alignmentOf(ptr) >= sizeof(MyStruct))`. Since it is very unlikely that anybody would need alignment larger than ~4GB, this seems enough. Note that it can and does work on 64-bit pointers. If you have an usecase which would warrant expanding this, I will do it.